### DISCUSSION TITLE
Framework - Incremental package dependencies

### DISCUSSION THREAD

**Discussion by lagacep-ans** [ansys/pyfluent#72] [Ideas]
Right now when we setup pyfluent it will pick up:
 pyvista, pyvistaqt, PySide6
 ... but as feature grows, we'll maybe end up adding matplotlib, tqdm, etc...

- On some machines/environment this might be a challenge, or impossible
- User might prefer to load only a minimal PyFluent environment, with no extra services

Do we need to tackle this as multiple levels of install ( configurable setup.py ? ) ?   Work with sub-packages?

(Dont take this wrong, the new pyvista connection is great.  but some users might not want to install that package, and would want a direct access to the underlying surface data, to process by their own means)

**Comment by dnwillia-work** [comment]
Totally agree.

**Reply by akaszynski** [reply]
One way to get around this issue would be to only install the "minimum requirements" when installing the package, and then add in optional requirements that are only imported at runtime. For example, if you want to plot, you'd protect the import with

```
try:
    import pyvista
except ImportError:
    raise ImportError("Please install pyvista with `pip install pyvista` to use this feature")
```

You may have to build in a few more lazy imports, but this option tends to scale well vs creating 50 packages.

**Comment by dnwillia-work** [comment]
We turned this into issue #305

**Comment by dnwillia-work** [comment]
I suppose that we have addressed this by splitting the repositories.
